filecoin-proofs/src/api/internal.rs

```diff
@@ -1142,4 +1189,24 @@ mod tests {
     fn post_verify_test() {
         post_verify_aux(SectorClass::Test, BytesAmount::Max);
     }
+
+    #[test]
+    fn partition_layer_challenges_test() {
+        let f = |partitions| {
+            select_challenges(
+                partitions,
+                POREP_MINIMUM_CHALLENGES,
+                LAYERS,
+                TAPER_LAYERS,
+                TAPER,
+            )
+            .all_challenges()
+        };
+        // Update to ensure all supported PoRepProofPartitions options are represented here.
+        assert_eq!(vec![1, 1, 2, 2], f(usize::from(PoRepProofPartitions::Two)));
```

Comment:
SUGGESTION: See comment at `POREP_PROOF_PARTITION_CHOICES`.

Reply:
As the number of choices grows, we could change this test to iterate through `POREP_PROOF_PARTITION_CHOICES` if we feel that's valuable. The value of having these lines here is less to test them as to expose the calculated result explicitly in code for reference. (It does also give an opportunity to sanity check correctness — but that should be primarily handled by the explicit tests below — which are fairly sparse now but could be supplemented if necessary.)